tests/openedx_tagging/core/tagging/test_system_defined_models.py

```diff
@@ -82,122 +78,171 @@ class TestModelSystemDefinedTaxonomy(TestTagTaxonomyMixin, TestCase):
+        )
+
+    def test_lp_taxonomy_validation(self):
+        """
+        Test that the validation methods of the Learning Package Taxonomy are working
+        """
+        # Create a new LearningPackage - we know no Tag instances will exist for it yet.
+        valid_lp = self._create_learning_pkg(key="valid-lp", title="New Learning Packacge")
+        # The taxonomy can validate tags by value which we've defined as they 'key' of the LearningPackage:
+        assert self.lp_taxonomy.validate_value(self.learning_pkg_2.key) is True
+        assert self.lp_taxonomy.validate_value(self.learning_pkg_2.key) is True
+        assert self.lp_taxonomy.validate_value(valid_lp.key) is True
+        assert self.lp_taxonomy.validate_value("foo") is False
+        # The taxonomy can also validate tags by external_id, which we've defined as the UUID of the LearningPackage:
+        assert self.lp_taxonomy.validate_external_id(self.learning_pkg_2.uuid) is True
+        assert self.lp_taxonomy.validate_external_id(self.learning_pkg_2.uuid) is True
+        assert self.lp_taxonomy.validate_external_id(valid_lp.uuid) is True
+        assert self.lp_taxonomy.validate_external_id("ba11225e-9ec9-4a50-87ea-3155c7c20466") is False
+
+    def test_author_taxonomy_validation(self):
+        """
+        Test the validation methods of the Author Taxonomy (Author = User)
+        """
+        assert self.author_taxonomy.validate_value(self.user_1.username) is True
+        assert self.author_taxonomy.validate_value(self.user_2.username) is True
+        assert self.author_taxonomy.validate_value("not a user") is False
+        # And we can validate by ID if we want:
+        assert self.author_taxonomy.validate_external_id(str(self.user_1.id)) is True
+        assert self.author_taxonomy.validate_external_id(str(self.user_2.id)) is True
+        assert self.author_taxonomy.validate_external_id("8742590") is False
+
     @ddt.data(
-        (ModelSystemDefinedTaxonomy, NotImplementedError),
-        (ModelObjectTag, NotImplementedError),
-        (InvalidModelTaxonomy, AssertionError),
-        (UserSystemDefinedTaxonomy, None),
+        "validate_value", "tag_for_value", "validate_external_id", "tag_for_external_id",
     )
-    @ddt.unpack
-    def test_implementation_error(self, taxonomy_cls, expected_exception):
-        if not expected_exception:
-            assert taxonomy_cls()
-        else:
-            with self.assertRaises(expected_exception):
-                taxonomy_cls()
-
-    # FIXME: something is wrong with this test case. It's setting the string
-    # "tag_id" as the primary key (integer) of the Tag instance, and it mentions
-    # "parent validation" but there is nothing to do with parents here.
-    #
-    # @ddt.data(
-    #     ("1", "tag_id", True),  # Valid
-    #     ("0", "tag_id", False),  # Invalid user
-    #     ("test_id", "tag_id", False),  # Invalid user id
-    #     ("1", None, False),  # Testing parent validations
```

Comment:
@bradenmacdonald  This case refers more to call the [super() functions](https://github.com/openedx/openedx-learning/pull/87/files#diff-8640a8393169a45e9926177fd6f2e8c7b0ad8439c96554334b95bbe912f52203L304) to call the parent class validations. Since that logic has been removed, I think it is okay to delete this test

Reply:
OK. Yeah for now I deleted it but put in some other new related tests.